### ISSUE TITLE
Paths mapping not working in Windows

### ISSUE THREAD

**Issue by alejandroclaro** [wessberg/rollup-plugin-ts#72]
Configuration with path mapping like this:

``` json
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDirs": [ "./src", "./tests" ],
    "baseUrl": ".",
    "paths": {
      "@src/*": [ "src/*" ],
      "@tests/*": [ "tests/*" ]
    }
  },
  "include": [
    "./src/**/*"
  ],
  "exclude": [
    "./node_modules",
    "./dist"
  ]
}
```

and the tsconfig.base.json:

``` json
{
  "compilerOptions": {
    "target": "es2018",
    "module": "esnext",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "composite": true,
    "declaration": true,
    "declarationMap": true,
    "experimentalDecorators": true,
    "incremental": true,
    "inlineSources": false,
    "inlineSourceMap": false,
    "listEmittedFiles": true,
    "noEmitOnError": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "removeComments": true,
    "sourceMap": true,
    "strict": true
  }
}
```
works in linux. But fails in windows. Once the use of "@src" and "@tests" is removed from the imports in the source code, the building processes is successful.

The error reported is:

```
(plugin Typscript) TS6059: File '...' is not under 'rootDir' '...'. 'rootDir' is expected to contain all source files.
```

As a side note. This is happening in a monorepo using yarn workspace.

**Comment by wessberg** [comment]
Hey there. Can you provide a stack trace or some more information so that I can see _when_ in the compilation that the error occurs? Several path mapping related tests are passing on Windows, so it must be related in some shape or form to a combination of a root directory outside of the current working directory and path mapping, but I'm guessing here.

**Comment by alejandroclaro** [comment]
@wessberg Thank you for your quick response.

I created an example here:

https://github.com/alejandroclaro/rollup-plugin-ts-issue-72

After creating this, I noticed that changing the order of the rollup plugins, solves the problem. Looks like a combination of @rollup/plugin-alias and rollup-plugin-ts. If the @rollup/plugin-alias is first in the list of plugins then I get the error when using path mapping in the tscofing.

This is the error output:

``` bash
$ yarn build
yarn run v1.21.1
$ lerna run build --stream
lerna notice cli v3.20.2
lerna info Executing command in 1 package: "yarn run build"
@company/my-package: $ rollup --environment NODE_ENV:development --config rollup.config.js
@company/my-package:
@company/my-package: ./src/index.ts → dist/rollup-plugin-ts-bug.cjs.js...
@company/my-package: [!] (plugin Typescript) TS6059: File '/Users/alejandro.claro/rollup-plugin-ts-issue-72/packages/my-package/src/foo.ts' is not under 'rootDir' 'C:/Users/alejandro.claro/rollup-plugin-ts-issue-72/packages/my-package'. 'rootDir' is expected to contain all source files.
@company/my-package: error Command failed with exit code 1.
@company/my-package: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run build exited 1 in '@company/my-package'
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

It seems odd to me that it work either way in Linux, but not in windows.

**Comment by alejandroclaro** [comment]
@wessberg Don't know if you can do something about the integration with @rollup/plugin-alias to make this work both in windows and Linux no matter the order of the plugins, but in case you cannot or decide to not do any workaround for this, I think it could be a good idea to include this in the FAQ section in the README.md or in a known issues section of the documentation.

For now, this has been solved by arranging the order of the plugins in our case.

**Comment by wessberg** [comment]
Hey there. I've been trying to investigate this bug and have made [a testcase](https://github.com/wessberg/rollup-plugin-ts/blob/master/test/integration.test.ts#L6) that tests placing `@rollup/plugin-alias` before `rollup-plugin-ts`. Thing is, that test passes on Windows. So, since I haven't been able to reproduce the bug you are experiencing with a test case, I've spun up a Windows test environment where I'll be able to see if I can understand what is happening, and if it is related to `rollup-plugin-ts` or `@rollup-plugin-alias`. Path mapping in general definitely works on Windows, so there must be some smoking gun somewhere in your repro which I'm looking forward to testing on a Windows machine.

**Comment by alejandroclaro** [comment]
@wessberg Thanks.

Have you been able to reproduce it using the example I did?

https://github.com/alejandroclaro/rollup-plugin-ts-issue-72

I have been able to reproduce the issue in different windows environments and also make sure it's alway working in Linux and MacOS. In windows, I was able to fix it by changing the order in my project and also in the example.

**Comment by wessberg** [comment]
I will clone your repro and test it soon in a Windows environment. As stated, I've already set up a test case based on your repro that does pass on Windows, so it will require some digging. I will report back when I know more 😊

**Comment by wessberg** [comment]
I'm seeing what's going on. For some reason, `@rollup/plugin-alias` rewrites the path into `\Users\...\packages\my-package\src\foo.ts` rather than `C:\Users\...\packages\my-package\src\foo.ts` (where C: could be any drive letter) which TypeScript doesn't like since it expects the path to include the drive letter. This is not really a bug in `rollup-plugin-ts` as the path is malformed according to TypeScript.

I added some internal logging to see what is passed from Rollup to the `transform` hook:

```
./src/index.ts → dist/rollup-plugin-ts-bug.cjs.js...
start: C:\Users\Frede\code\repro\rollup-plugin-ts-issue-72\packages\my-package\src\index.ts
start: \Users\Frede\code\repro\rollup-plugin-ts-issue-72\packages\my-package\src\foo.ts
[!] (plugin Typescript) TS6059: File '/Users/Frede/code/repro/rollup-plugin-ts-issue-72/packages/my-package/src/foo.ts' is not under 'rootDir' 'C:/Users/Frede/code/repro/rollup-plugin-ts-issue-72/packages/my-package'. 'rootDir' is expected to contain all source files.
```

The reason why it works when `@rollup/plugin-alias` is located _after_ `rollup-plugin-ts` - or even if you leave that plugin out entirely! - is because `rollup-plugin-ts` itself resolves aliased paths and returns the correct ones to Rollup.

Anyway, I'd like for this to work, so I've added logic for ensuring that absolute paths on Windows always include a drive letter before passing them to TypeScript.

It has been released in [v1.2.16](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.2.16)
